src/pdc-api.ts

```diff
@@ -50,6 +50,101 @@ const usePdcApi = <T>(
   return response;
 };
 
+const usePdcCallbackApi = <T>(
+  path: string,
+): (options: RequestInit) => Promise<T> => {
+  const { fetch } = useOidcFetch();
+
+  return useCallback((options: RequestInit) => {
+    const url = new URL(path, API_URL);
+    return fetch(url, options)
+      .then(throwNotOk)
+      .then((res) => res.json())
+      .catch((e) => logError(e, path, options));
+
+    /* eslint-disable-next-line react-hooks/exhaustive-deps --
+     *
+     * fetch should not be a dependency, because although it or its internal
+     * state may change from render to render, such changes are not relevant:
+     * a change in the way we make a request should not trigger an API request
+     */
+  }, [path]);
+};
+
+interface ApiPresignedPostRequest {
+  fileType: string;
+  fileSize: number;
+}
+
+interface ApiPresignedPostResponse {
+  fileType: string;
+  fileSize: number;
+  presignedPost: {
+    url: string;
+    fields: Record<string, string> & {
+      key: string,
+    };
+  };
+}
+
+const usePresignedPostCallback = () => {
+  const api = usePdcCallbackApi<ApiPresignedPostResponse>('/presignedPostRequests');
+  return (params: ApiPresignedPostRequest) => api({
+    method: 'post',
+    headers: {
+      Accept: 'application/json',
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify(params),
+  });
+};
+
+const uploadPresignedPost = (
+  post: ApiPresignedPostResponse,
+  file: File,
+) => {
+  const formData = new FormData();
+  Object.entries(post.presignedPost.fields).forEach(([key, value]) => (
+    formData.append(key, value)
+  ));
+  formData.append('Content-Type', post.fileType);
+  formData.append('file', file);
+  return fetch(
+    post.presignedPost.url,
+    {
+      method: 'post',
+      body: formData,
+    },
+  )
+    .then(throwNotOk)
```

Comment:
Same here? Why `throwNotOk`?

Reply:
nm, actually read the source for the function. I object to the unclear name!

Reply:
I bet that is a confusingly named helper that means  "throw if not OK"

I'm gonna go ahead and confirm that bet, and then we should update it to something more clear.

Reply:
```
const throwNotOk = (res: Response): Response => {
  if (!res.ok) {
    throw new Error(`Response status: ${res.status} ${res.statusText}`);
  }
  return res;
};
```

My bet was right!  I'll change that in a separate commit.